### PR TITLE
Make test parametrization override indirect fixtures

### PR DESCRIPTION
**Pull request by egnartsms**
A fix for https://github.com/pytest-dev/pytest/issues/2220.

### CODE REVIEW

File: src/_pytest/fixtures.py [hunk 1]
```diff
@@ -274,11 +274,45 @@ def get_direct_param_fixture_func(request):
...
+            # FixtureDefs which wrap 'get_direct_param_fixture_func(request)'.
+            # So they introduce the new dependency 'request' which might have
+            # been missing in the original tree (closure).
+            if argname not in closure and argname in self.names_closure:
+                closure.add(argname)
+                if argname in self.name2fixturedefs:
+                    working_set.update(self.name2fixturedefs[argname][-1].argnames)
+
+        self.names_closure[:] = sorted(
+            closure, key=lambda name: self.names_closure.index(name)
```
Comment (RonnyPfannschmidt) [inline src/_pytest/fixtures.py]: the key could just be `self.names_closure.index`

**Comment by RonnyPfannschmidt** [comment]
since this one corrects a rather strange misbehavior, i wonder if its better to target the feature branch just to avoid certain kinds of disruption in a patch release

CC @nicoddemus

**Comment by egnartsms** [comment]
@RonnyPfannschmidt I added those type definitions as a form of documentation, just to make it clear what the fields of FuncFixtureInfo are. Not sure what you mean by tooling.

**Comment by nicoddemus** [comment]
> since this one corrects a rather strange misbehavior, i wonder if its better to target the feature branch just to avoid certain kinds of disruption in a patch release

I haven't had a chance to review the PR yet, but I agree that even if this changes a somewhat wrong behavior, we better be safe and add this to the next feature release.

**Comment by RonnyPfannschmidt** [comment]
we will have to do our own homework first and merge master to features before this one can apply as intended

**Comment by RonnyPfannschmidt** [comment]
#3634 will sort the conflict out then i will try again,

**Comment by RonnyPfannschmidt** [comment]
@egnartsms oh, i missed it before, can you please add a news fragment about the bugfix

once thats complete its good to merge :+1:

**Comment by egnartsms** [comment]
@RonnyPfannschmidt do you mean to create a file in the changelog directory? Or something else? That's my first contribution to pytest.

**Comment by RonnyPfannschmidt** [comment]
@edisongustavo correct a file in the changelog directory

**Pull request merged**